packages/ag-charts-community/src/chart/data/dataModel.ts

```diff
@@ -429,6 +442,78 @@ export class DataModel<
         return column;
     }
 
+    *forEachDatum(scope: ScopeProvider, processedData: GroupedData<any>, group: DataGroup) {
```

Comment:
These `forEachX` iterator methods are to abstract the `Series` from the data-structure slightly, as a pre-emptive attempt to avoid data-structure optimisation from requiring series changes.

Reply:
Added